### PR TITLE
Make pdata.LogsToOtlp and pdata.LogsFromOtlp impossible to call from outside this module

### PR DESCRIPTION
**Pull request by tigrannajaryan**
LogsToOtlp and LogsFromOtlp are intended to be used only by OTLP exporter (for which
OTLP is the native data format) and fileexporter (for which we do want to output internal
representation for debugging purposes).

All other code is supposed to use public member functions of pdata.Logs struct.

This changes makes it impossible to use LogsToOtlp and LogsFromOtlp outside this module
by hiding them behind an intermediary struct that is declared in an internal package.

This is necessary to prevent accidental use of LogsToOtlp and LogsFromOtlp and the OTLP
data structs, which we prohibit to do since the OTLP data structs may change in the
future.

### CODE REVIEW

File: consumer/pdata/timestamp.go
Comment (bogdandrutu) [inline consumer/pdata/timestamp.go:15]: This can be extracted in a separate PR

Comment (tigrannajaryan) [reply consumer/pdata/timestamp.go]: Done in https://github.com/open-telemetry/opentelemetry-collector/pull/1709

**Comment by tigrannajaryan** [comment]
@open-telemetry/collector-approvers I need a review/approval on this.

**Review by dmitryax — APPROVED** [review]
LGTM

**Pull request merged**
